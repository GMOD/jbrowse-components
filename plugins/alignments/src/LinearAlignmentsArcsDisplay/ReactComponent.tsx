import React from 'react'

export default function () {
  return <div>Hello world</div>
}
