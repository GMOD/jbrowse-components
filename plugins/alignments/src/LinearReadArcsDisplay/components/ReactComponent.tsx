import React, { useCallback } from 'react'
import { getContainingView } from '@jbrowse/core/util'
import { observer } from 'mobx-react'

// local
import BaseDisplayComponent from '../../shared/components/BaseDisplayComponent'
import type { LinearReadArcsDisplayModel } from '../model'
import type { LinearGenomeViewModel } from '@jbrowse/plugin-linear-genome-view'

type LGV = LinearGenomeViewModel

const Arcs = observer(function ({
  model,
}: {
  model: LinearReadArcsDisplayModel
}) {
  const view = getContainingView(model) as LGV
  const width = Math.round(view.dynamicBlocks.totalWidthPx)
  const height = model.height
  // biome-ignore lint/correctness/useExhaustiveDependencies:
  const cb = useCallback(
    (ref: HTMLCanvasElement) => {
      model.setRef(ref)
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [model, width, height],
  )

  // note: the position absolute below avoids scrollbar from appearing on track
  return (
    <canvas
      data-testid="arc-canvas"
      ref={cb}
      style={{ width, height, position: 'absolute' }}
      width={width * 2}
      height={height * 2}
    />
  )
})

const LinearReadArcsReactComponent = observer(function ({
  model,
}: {
  model: LinearReadArcsDisplayModel
}) {
  return (
    <BaseDisplayComponent model={model}>
      <Arcs model={model} />
    </BaseDisplayComponent>
  )
})

export default LinearReadArcsReactComponent
