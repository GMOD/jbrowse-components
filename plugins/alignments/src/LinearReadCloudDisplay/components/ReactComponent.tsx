import React, { useCallback } from 'react'
import { getContainingView } from '@jbrowse/core/util'
import { observer } from 'mobx-react'

// local
import BaseDisplayComponent from '../../shared/components/BaseDisplayComponent'
import type { LinearReadCloudDisplayModel } from '../model'
import type { LinearGenomeViewModel } from '@jbrowse/plugin-linear-genome-view'

type LGV = LinearGenomeViewModel

const Cloud = observer(function ({
  model,
}: {
  model: LinearReadCloudDisplayModel
}) {
  const view = getContainingView(model) as LGV
  const width = Math.round(view.dynamicBlocks.totalWidthPx)
  const height = model.height

  // biome-ignore lint/correctness/useExhaustiveDependencies:
  const cb = useCallback(
    (ref: HTMLCanvasElement) => {
      model.setRef(ref)
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [model, width, height],
  )

  // note: the position absolute below avoids scrollbar from appearing on track
  return (
    <canvas
      data-testid="cloud-canvas"
      ref={cb}
      style={{ width, height, position: 'absolute' }}
      width={width * 2}
      height={height * 2}
    />
  )
})

const LinearReadCloudReactComponent = observer(function ({
  model,
}: {
  model: LinearReadCloudDisplayModel
}) {
  return (
    <BaseDisplayComponent model={model}>
      <Cloud model={model} />
    </BaseDisplayComponent>
  )
})
export default LinearReadCloudReactComponent
