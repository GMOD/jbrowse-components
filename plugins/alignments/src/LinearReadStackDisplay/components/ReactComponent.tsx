import { useCallback, useMemo, useRef, useState } from 'react'

import BaseTooltip from '@jbrowse/core/ui/BaseTooltip'
import { getContainingView } from '@jbrowse/core/util'
import Flatbush from '@jbrowse/core/util/flatbush'
import { observer } from 'mobx-react'

import BaseDisplayComponent from '../../shared/components/BaseDisplayComponent'

import type { ReducedFeature } from '../../shared/fetchChains'
import type { LinearReadStackDisplayModel } from '../model'
import type { LinearGenomeViewModel } from '@jbrowse/plugin-linear-genome-view'

type LGV = LinearGenomeViewModel

const Stack = observer(function ({
  model,
}: {
  model: LinearReadStackDisplayModel
}) {
  const view = getContainingView(model) as LGV
  const width = Math.round(view.dynamicBlocks.totalWidthPx)
  const height = model.height
  const containerRef = useRef<HTMLDivElement>(null)
  const [hoveredFeature, setHoveredFeature] = useState<{
    x: number
    y: number
    width: number
    height: number
  } | null>(null)
  const [hoveredFeatureData, setHoveredFeatureData] = useState<ReducedFeature | null>(null)
  const [mousePosition, setMousePosition] = useState<{ x: number; y: number } | null>(null)

  // Convert flatbush data to Flatbush instance
  const flatbushIndex = useMemo(() => {
    return model.featureLayout ? Flatbush.from(model.featureLayout.data) : null
  }, [model.featureLayout])

  // biome-ignore lint/correctness/useExhaustiveDependencies:
  const cb = useCallback(
    (ref: HTMLCanvasElement) => {
      model.setRef(ref)
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [model, width, height],
  )

  // biome-ignore lint/correctness/useExhaustiveDependencies:
  const mouseoverCb = useCallback(
    (ref: HTMLCanvasElement) => {
      model.setMouseoverRef(ref)
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [model, width, height],
  )

  const onMouseMove = useCallback(
    (event: React.MouseEvent) => {
      if (!containerRef.current || !flatbushIndex) {
        setHoveredFeature(null)
        setHoveredFeatureData(null)
        setMousePosition(null)
        return
      }

      const rect = containerRef.current.getBoundingClientRect()
      const offsetX = event.clientX - rect.left
      const offsetY = event.clientY - rect.top

      // Track mouse position for tooltip
      setMousePosition({ x: event.clientX, y: event.clientY })

      // Search for features at this position
      const results = flatbushIndex.search(offsetX, offsetY, offsetX + 1, offsetY + 1)

      if (results.length > 0) {
        const featureIndex = results[0]!
        const feature = model.featuresForFlatbush[featureIndex]

        if (feature) {
          setHoveredFeature({
            x: feature.x1,
            y: feature.y1,
            width: feature.x2 - feature.x1,
            height: feature.y2 - feature.y1,
          })
          setHoveredFeatureData(feature.data)
        } else {
          setHoveredFeature(null)
          setHoveredFeatureData(null)
        }
      } else {
        setHoveredFeature(null)
        setHoveredFeatureData(null)
      }
    },
    [flatbushIndex, model.featuresForFlatbush],
  )

  const onMouseLeave = useCallback(() => {
    setHoveredFeature(null)
    setHoveredFeatureData(null)
    setMousePosition(null)
  }, [])

  // note: the position absolute below avoids scrollbar from appearing on track
  return (
    <div
      ref={containerRef}
      style={{ position: 'relative', width, height }}
      onMouseMove={onMouseMove}
      onMouseLeave={onMouseLeave}
    >
      <canvas
        data-testid="stack-canvas"
        ref={cb}
        style={{ width, height, position: 'absolute', left: 0, top: 0 }}
        width={width * 2}
        height={height * 2}
      />
      <canvas
        data-testid="stack-mouseover-canvas"
        ref={mouseoverCb}
        style={{ width, height, position: 'absolute', left: 0, top: 0, pointerEvents: 'none' }}
        width={width * 2}
        height={height * 2}
      />
      {hoveredFeature ? (
        <div
          style={{
            position: 'absolute',
            left: hoveredFeature.x,
            top: hoveredFeature.y,
            width: hoveredFeature.width,
            height: hoveredFeature.height,
            backgroundColor: 'rgba(0, 0, 0, 0.2)',
            pointerEvents: 'none',
          }}
        />
      ) : null}
      {hoveredFeatureData && mousePosition ? (
        <BaseTooltip clientPoint={{ x: mousePosition.x + 15, y: mousePosition.y }}>
          <div>
            <div><strong>{hoveredFeatureData.name}</strong></div>
            <div>{hoveredFeatureData.refName}:{hoveredFeatureData.start.toLocaleString()}-{hoveredFeatureData.end.toLocaleString()}</div>
            <div>Strand: {hoveredFeatureData.strand === 1 ? '+' : hoveredFeatureData.strand === -1 ? '-' : '.'}</div>
            {hoveredFeatureData.flags !== undefined ? <div>Flags: {hoveredFeatureData.flags}</div> : null}
            {hoveredFeatureData.tlen !== undefined && hoveredFeatureData.tlen !== 0 ? <div>Template length: {hoveredFeatureData.tlen}</div> : null}
          </div>
        </BaseTooltip>
      ) : null}
    </div>
  )
})

const LinearReadStackReactComponent = observer(function ({
  model,
}: {
  model: LinearReadStackDisplayModel
}) {
  return (
    <BaseDisplayComponent model={model}>
      <Stack model={model} />
    </BaseDisplayComponent>
  )
})
export default LinearReadStackReactComponent
