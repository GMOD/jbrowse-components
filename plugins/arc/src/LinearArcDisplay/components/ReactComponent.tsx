import React from 'react'
import { observer } from 'mobx-react'
import { getContainingView } from '@jbrowse/core/util'
import { LinearGenomeViewModel } from '@jbrowse/plugin-linear-genome-view'

// local
import { LinearArcDisplayModel } from '../model'
import BaseDisplayComponent from '../../shared/BaseDisplayComponent'

type LGV = LinearGenomeViewModel

const Arcs = observer(function ({ model }: { model: LinearArcDisplayModel }) {
  const view = getContainingView(model) as LGV
  const width = Math.round(view.dynamicBlocks.totalWidthPx)
  const height = model.height

  // note: the position absolute below avoids scrollbar from appearing on track
  return (
    <svg
      data-testid="arc-canvas"
      style={{ width, height, position: 'absolute' }}
      width={width * 2}
      height={height * 2}
    />
  )
})

const LinearArcReactComponent = observer(function ({
  model,
}: {
  model: LinearArcDisplayModel
}) {
  return (
    <BaseDisplayComponent model={model}>
      <Arcs model={model} />
    </BaseDisplayComponent>
  )
})

export default LinearArcReactComponent
