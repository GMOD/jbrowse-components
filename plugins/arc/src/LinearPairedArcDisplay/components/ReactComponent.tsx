import React from 'react'
import { observer } from 'mobx-react'

// local
import Arcs from './Arcs'
import BaseDisplayComponent from './BaseDisplayComponent'
import type { LinearArcDisplayModel } from '../model'

const LinearArcReactComponent = observer(function ({
  model,
  exportSVG,
}: {
  model: LinearArcDisplayModel
  exportSVG?: boolean
}) {
  return (
    <BaseDisplayComponent model={model}>
      <Arcs model={model} exportSVG={exportSVG} />
    </BaseDisplayComponent>
  )
})

export default LinearArcReactComponent
