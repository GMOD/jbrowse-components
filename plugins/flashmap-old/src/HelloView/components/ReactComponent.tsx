import React, { useState } from 'react'

export default function ReactComponent() {
  const [pushed, setPushed] = useState('')
  return (
    <div style={{ padding: 50 }}>
      <h1>Hello plugin developers!</h1>
      <button
        onClick={() => {
          setPushed('Whoa! You pushed the button!')
        }}
      >
        Push the button
      </button>
      <p>{pushed}</p>
    </div>
  )
}
