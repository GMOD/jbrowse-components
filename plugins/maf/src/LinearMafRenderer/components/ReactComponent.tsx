import React from 'react'

import { PrerenderedCanvas } from '@jbrowse/core/ui'
import { observer } from 'mobx-react'

const LinearMafRendering = observer(function (props: {
  width: number
  height: number
}) {
  return <PrerenderedCanvas {...props} />
})

export default LinearMafRendering
