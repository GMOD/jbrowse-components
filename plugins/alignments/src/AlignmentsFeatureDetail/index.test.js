import { render } from '@testing-library/react'
import { types } from 'mobx-state-tree'
import React from 'react'
import PluginManager from '@jbrowse/core/PluginManager'
import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { stateModelFactory } from '.'
import ReactComponent from './AlignmentsFeatureDetail'

test('open up a widget', () => {
  console.warn = jest.fn()
  const pluginManager = new PluginManager([])

  const Session = types.model({
    pluginManager: types.optional(types.frozen(), {}),
    rpcManager: types.optional(types.frozen(), {}),
    configuration: ConfigurationSchema('test', {}),
    widget: stateModelFactory(pluginManager),
  })
  const session = Session.create({
    widget: { type: 'AlignmentsFeatureWidget' },
  })
  session.widget.setFeatureData({
    seq: 'TTGTTGCGGAGTTGAACAACGGCATTAGGAACACTTCCGTCTCTCACTTTTATACGATTATGATTGGTTCTTTAGCCTTGGTTTAGATTGGTAGTAGTAG',
    start: 2,
    end: 102,
    strand: 1,
    score: 37,
    qual: '17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17',
    MQ: 37,
    CIGAR: '100M',
    length_on_ref: 100,
    template_length: 0,
    seq_length: 100,
    name: 'ctgA_3_555_0:0:0_2:0:0_102d',
    refName: 'ctgA',
    type: 'match',
  })
  const { container, getByText } = render(
    <ReactComponent model={session.widget} />,
  )
  expect(container.firstChild).toMatchSnapshot()
  expect(getByText('ctgA:3..102 (+)')).toBeTruthy()
})
