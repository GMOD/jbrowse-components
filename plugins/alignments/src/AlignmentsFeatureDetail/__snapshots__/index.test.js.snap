// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`open up a widget 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  data-testid="alignment-side-drawer"
>
  <div
    class="MuiPaper-root makeStyles-paperRoot MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiPaper-root MuiAccordion-root MuiAccordion-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      style="margin-top: 4px;"
    >
      <div
        aria-disabled="false"
        aria-expanded="true"
        class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-expanded"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content MuiAccordionSummary-expanded"
        >
          <span
            class="MuiTypography-root MuiTypography-button"
          >
             
            Primary data
          </span>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiAccordionSummary-expanded MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-expandIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-entered"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiAccordionDetails-root makeStyles-expansionPanelDetails"
              >
                <div
                  class="makeStyles-field"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Position
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      ctgA:3..102 (+)
                    </div>
                  </div>
                </div>
                <div
                  class="makeStyles-field"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Name
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      ctgA_3_555_0:0:0_2:0:0_102d
                    </div>
                  </div>
                </div>
                <div
                  class="makeStyles-field"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Length
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      100
                    </div>
                  </div>
                </div>
                <div
                  class="makeStyles-field"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    Type
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      match
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiPaper-root MuiAccordion-root MuiAccordion-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
      style="margin-top: 4px;"
    >
      <div
        aria-disabled="false"
        aria-expanded="true"
        class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-expanded"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content MuiAccordionSummary-expanded"
        >
          <span
            class="MuiTypography-root MuiTypography-button"
          >
             
            Attributes
          </span>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiAccordionSummary-expanded MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-expandIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-entered"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiAccordionDetails-root makeStyles-expansionPanelDetails"
              >
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    seq
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      TTGTTGCGGAGTTGAACAACGGCATTAGGAACACTTCCGTCTCTCACTTTTATACGATTATGATTGGTTCTTTAGCCTTGGTTTAGATTGGTAGTAGTAG
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    score
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      37
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    qual
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <button
                      type="button"
                    >
                      Copy
                    </button>
                    <button
                      type="button"
                    >
                      Show more
                    </button>
                    <div>
                      17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 17 1...
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    MQ
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      37
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    CIGAR
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      100M
                    </div>
                  </div>
                </div>
                <div
                  style="display: flex;"
                >
                  <div
                    class="makeStyles-fieldName"
                  >
                    seq_length
                  </div>
                  <div
                    class="makeStyles-fieldValue"
                  >
                    <div>
                      100
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiAccordion-root MuiAccordion-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    style="margin-top: 4px;"
  >
    <div
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-expanded"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content MuiAccordionSummary-expanded"
      >
        <span
          class="MuiTypography-root MuiTypography-button"
        >
           
          Flags
        </span>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiAccordionSummary-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-expandIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-container MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            role="region"
          >
            <div
              class="MuiAccordionDetails-root makeStyles-expansionPanelDetails"
            >
              <div
                style="display: flex;"
              >
                <div
                  class="makeStyles-fieldName"
                >
                  Flag
                </div>
                <div
                  class="makeStyles-fieldValue"
                />
              </div>
              <div>
                <input
                  id="read paired_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="read paired_0"
                >
                  read paired
                </label>
              </div>
              <div>
                <input
                  id="read mapped in proper pair_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="read mapped in proper pair_0"
                >
                  read mapped in proper pair
                </label>
              </div>
              <div>
                <input
                  id="read unmapped_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="read unmapped_0"
                >
                  read unmapped
                </label>
              </div>
              <div>
                <input
                  id="mate unmapped_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="mate unmapped_0"
                >
                  mate unmapped
                </label>
              </div>
              <div>
                <input
                  id="read reverse strand_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="read reverse strand_0"
                >
                  read reverse strand
                </label>
              </div>
              <div>
                <input
                  id="mate reverse strand_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="mate reverse strand_0"
                >
                  mate reverse strand
                </label>
              </div>
              <div>
                <input
                  id="first in pair_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="first in pair_0"
                >
                  first in pair
                </label>
              </div>
              <div>
                <input
                  id="second in pair_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="second in pair_0"
                >
                  second in pair
                </label>
              </div>
              <div>
                <input
                  id="not primary alignment_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="not primary alignment_0"
                >
                  not primary alignment
                </label>
              </div>
              <div>
                <input
                  id="read fails platform/vendor quality checks_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="read fails platform/vendor quality checks_0"
                >
                  read fails platform/vendor quality checks
                </label>
              </div>
              <div>
                <input
                  id="read is PCR or optical duplicate_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="read is PCR or optical duplicate_0"
                >
                  read is PCR or optical duplicate
                </label>
              </div>
              <div>
                <input
                  id="supplementary alignment_0"
                  readonly=""
                  type="checkbox"
                />
                <label
                  for="supplementary alignment_0"
                >
                  supplementary alignment
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
